src/cdk/overlay/overlay-directives.ts

```diff
@@ -179,6 +186,21 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
   get lockPosition() { return this._lockPosition; }
   set lockPosition(value: any) { this._lockPosition = coerceBooleanProperty(value); }
 
+  /** Whether the overlay's width and height can be constrained to fit within the viewport. */
+  @Input('cdkConnectedOverlayFlexibleDimensions')
+  get flexibleDiemsions() { return this._flexibleDimensions; }
+  set flexibleDiemsions(value: any) { this._flexibleDimensions = coerceBooleanProperty(value); }
+
+  /** Whether the overlay can grow after the initial open when flexible positioning is turned on. */
+  @Input('cdkConnectedOverlayGrowAfterOpen')
+  get growAfterOpen() { return this._growAfterOpen; }
+  set growAfterOpen(value: any) { this._growAfterOpen = coerceBooleanProperty(value); }
+
+  /** Whether the overlay can be pushed on-screen if none of the provided positions fit. */
+  @Input('cdkConnectedOverlayGrowAfterOpen')
```

Comment:
Add a unit test that adds these just to ensure they don't break if e.g. someone accidentally renames a input for something minor like that?